internal/util/helpers_test.go

```diff
@@ -4,18 +4,758 @@
+			if err != nil {
+				t.Errorf("error setting XDG_CONFIG_HOME: %v", err)
+			}
+			path, err := util.GetConfigDirPath()
+			if (err != nil) != tt.expectingError {
+				t.Errorf("expected error: %v, got: %v", tt.expectingError, err)
+			}
+			if path != tt.expectedPath {
+				t.Errorf("expected path: %s, got: %s", tt.expectedPath, path)
+			}
+		})
+	}
+}
+
+// TestGetGrpcConnection tests the GetGrpcConnection function
+func TestGetGrpcConnection(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		grpcHost      string
+		grpcPort      int
+		allowInsecure bool
+		issuerUrl     string
+		clientId      string
+		envToken      string
+		expectedError bool
+	}{
+		{
+			name:          "Valid GRPC connection to localhost with secure connection",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: false,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to localhost with insecure connection",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: true,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to localhost without passing MINDER_AUTH_TOKEN as an argument",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: false,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server with secure connection",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: false,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server with insecure connection",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: true,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server without passing MINDER_AUTH_TOKEN as an argument",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: true,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "",
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			originalEnvToken := os.Getenv(util.MinderAuthTokenEnvVar)
+			err := os.Setenv(util.MinderAuthTokenEnvVar, tt.envToken)
+			// reset this the environment variable when complete.
+			defer os.Setenv(util.MinderAuthTokenEnvVar, originalEnvToken)
```

Comment:
You'll want to protect this with a mutex as well.  Given the likely test execution time, I'd probably be lazy and re-use `envLock` from `TestGetConfigDirPath`,  but you could also use separate locks for each environment variable if you're feeling particularly ambitious (i wouldn't recommend it, though).